src/transformState.js

```diff
@@ -5,7 +5,34 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  const finiteStateMachine = ({ type, extraData, keysToRemove }) => {
+    switch (type) {
+      case 'addProperties':
+        Object.assign(state, extraData);
+        break;
+
+      case 'removeProperties':
+        keysToRemove.forEach(keyToRemove => {
+          delete state[keyToRemove];
+        });
+        break;
+
+      case 'clear':
+        for (const key in state) {
+          delete state[key];
+        }
+        break;
+
+      default:
+        break;
```

Comment:
![image](https://github.com/mate-academy/js_stateful-object/assets/54860641/d7477526-b1c3-4669-b3b3-d6b8d939306e)
For this task, you can return some message that something went wrong.